File: solution/27965/main.py
def main():
    input = open(0, "rb").readline
    N, K = map(int, input().split())
    remainder = 0
    ten = 10
    for i in range(1, N + 1):
        if i == ten:
            ten *= 10
        remainder = (remainder * ten + i) % K
    print(remainder)


if __name__ == "__main__":
    main()

File: solution/27965/Main.java
import java.io.*;

public class Main {
    static BufferedReader stdin = new BufferedReader(new InputStreamReader(System.in), 1048576);
    static StringBuilder stdout = new StringBuilder();
    public static void main(String[] args) throws Exception {
        String[] line = stdin.readLine().split(" ");
        long N = Integer.parseInt(line[0]);
        int K = Integer.parseInt(line[1]);
        long remainder = 0, ten = 10;
        for (long i = 1; i <= N; ++i) {
            if (i == ten) {
                ten *= 10;
            }
            remainder = (remainder * ten + i) % K;
        }
        stdout.append(remainder);
        System.out.println(stdout);
    }
}
